Diamond segments shall not be joined with text-only segments

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

My prompt is set up to look like this:
![omp_04](https://github.com/JanDeDobbeleer/oh-my-posh/assets/38878653/3e0122bc-80c7-4837-83a6-c26b20ba9720)
(computer and user name has been masked).

However with the version 19.11.06 it now looks like this:
![omp_06](https://github.com/JanDeDobbeleer/oh-my-posh/assets/38878653/15b46597-5f35-41e6-99ae-3cbb1ab1848d)

The "SH" part is a text-only segment so it does not have a trailing diamond character.

The reason why I see this problem on Linux is that on this platform the updates of OMP get out a bit faster (I use homebrew). On Windows I get the updates through chocolatey and here there may be a (usually) short delay.

### Theme

Added as zip file because I cannot upload a .yml file...
[ekologic.omp.zip](https://github.com/JanDeDobbeleer/oh-my-posh/files/14445124/ekologic.omp.zip)


### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
Version: 19.11.6

Shell: bash (5.1.16(1)-release)

Prompt:


┌ username@COMPUTER SH ~
└ $ 

Segments:

ConsoleTitle(true)                        -   0 ms
Session(true)                             -   2 ms
Shell(true)                               -   0 ms
Python(false)                             -   0 ms
Path(true)                                -   2 ms
Status(false)                             -   2 ms
Text(false)                               -   0 ms
Project(false)                            -   1 ms
Gitversion(false)                         - 229 ms
Git(false)                                -   1 ms
Svn(false)                                -   1 ms
Root(false)                               -   0 ms
Text(true)                                -   0 ms
Text(true)                                -   0 ms

Run duration: 239.30568ms

Cache path: /home/username/.cache/oh-my-posh

Config path: /home/username/ekologic.omp.yml

Logs:

[DEBUG] 09:38:35.397 shell.go:Getenv:398 → NO DATA
[TRACE] 09:38:35.397 shell.go:Getenv(TERM_PROGRAM) - 22.5µs
[DEBUG] 09:38:35.397 shell.go:Getenv:398 → NO DATA
[TRACE] 09:38:35.397 shell.go:Getenv(OMP_CACHE_DIR) - 2.8µs
[DEBUG] 09:38:35.397 shell.go:Getenv:398 → NO DATA
[TRACE] 09:38:35.397 shell.go:Getenv(XDG_CACHE_HOME) - 2.1µs
[TRACE] 09:38:35.397 shell_unix.go:CachePath() - 19µs
[DEBUG] 09:38:35.397 shell.go:Getenv:398 → /home/username/ekologic.omp.yml
[TRACE] 09:38:35.397 shell.go:Getenv(POSH_THEME) - 4.8µs
[DEBUG] 09:38:35.397 shell_unix.go:Platform:87 → ubuntu
[TRACE] 09:38:35.397 shell.go:resolveConfigPath() - 12.6µs
[TRACE] 09:38:35.397 shell.go:Init() - 146.5µs
[TRACE] 09:38:35.397 shell.go:Flags() - 0s
[TRACE] 09:38:35.402 config.go:loadConfig() - 4.477204ms
[TRACE] 09:38:35.402 shell.go:Flags() - 0s
[DEBUG] 09:38:35.402 shell.go:Getenv:398 → NO DATA
[TRACE] 09:38:35.402 shell.go:Getenv(OMP_CACHE_DISABLED) - 4.7µs
[DEBUG] 09:38:35.402 shell.go:Shell:663 → no shell name provided in flags, trying to detect it
[DEBUG] 09:38:35.402 shell.go:Shell:671 → process name: bash
[TRACE] 09:38:35.402 shell.go:Shell() - 418.5µs
[DEBUG] 09:38:35.402 shell.go:Getenv:398 → 5.1.16(1)-release
[TRACE] 09:38:35.402 shell.go:Getenv(POSH_SHELL_VERSION) - 3.5µs
[DEBUG] 09:38:35.402 debug.go:PrintDebug:22 → Segment: Title
[DEBUG] 09:38:35.402 text.go:Render:70 → Rendering template: {{.Env.PROJECT_NAME}}
[TRACE] 09:38:35.402 shell_unix.go:Root() - 1.3µs
[TRACE] 09:38:35.402 shell.go:Shell() - 100ns
[TRACE] 09:38:35.402 shell.go:StatusCodes() - 0s
[DEBUG] 09:38:35.402 shell_unix.go:IsWsl:35 → true
[TRACE] 09:38:35.402 shell_unix.go:IsWsl() - 3.7µs
[DEBUG] 09:38:35.402 shell.go:TemplateCache:844 ↓
    environment: [SHELL=/bin/bash LIBGL_ALWAYS_INDIRECT=1 WSL2_GUI_APPS_ENABLED=1 POSH_PID=7423 WSL_DISTRO_NAME=Ubuntu-22.04 PROJECT_NAME=No Project WT_SESSION=9edddd3b-eecc-46a1-8473-fd3fb321ae56 SSH_AUTH_SOCK=/tmp/ssh-XXXXXXx9BwpU/agent.598 HOMEBREW_PREFIX=/home/linuxbrew/.linuxbrew SSH_AGENT_PID=599 EDITOR=geany NAME=COMPUTER PWD=/home/username USERPROFILE=/home/username LOGNAME=username X_SERVER_INSTALLED=true MANPATH=/home/linuxbrew/.linuxbrew/share/man: MOTD_SHOWN=update-motd HOME=/home/username LANG=C.UTF-8 WSL_INTEROP=/run/WSL/7420_interop SSHKEYLIST=ekologic LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36: POSH_SHELL_VERSION=5.1.16(1)-release VIRTUAL_ENV=/home/username/.venv312 WAYLAND_DISPLAY=wayland-0 CONDA_PROMPT_MODIFIER=false WORK_CONTEXT=work INFOPATH=/home/linuxbrew/.linuxbrew/share/info: LESSCLOSE=/usr/bin/lesspipe %s %s TERM=xterm-256color POSH_CURSOR_COLUMN=1 LESSOPEN=| /usr/bin/lesspipe %s USER=username USERDOCUMENTS=/home/username/Documents HOMEBREW_CELLAR=/home/linuxbrew/.linuxbrew/Cellar SLICK_VERSION=slickedit-pro2023 SHLVL=1 HOMEBREW_REPOSITORY=/home/linuxbrew/.linuxbrew/Homebrew PROJECTS_BASE=/home/username/projects POSH_CURSOR_LINE=39 VIRTUAL_ENV_PROMPT=(.venv312)  POSH_THEME=/home/username/ekologic.omp.yml VIRTUAL_ENV_PATH=/home/username/.venv312 XDG_RUNTIME_DIR=/run/user/1000/ PS1=
    \[[38;2;117;117;117m\]┌ username@COMPUTER \[[0m\]\[[38;2;255;255;255m\]SH \[[0m\]\[[48;2;145;221;255m\]\[[38;2;145;221;255m\]\[[0m\]\[[48;2;145;221;255m\]\[[38;2;16;14;35m\]~\[[0m\]\[[38;2;145;221;255m\]\[[0m\]
    \[[38;2;117;117;117m\]└\[[0m\]\[[38;2;255;255;255m\] $\[[0m\]\[]0;No Project\]  WSLENV=WT_SESSION:WT_PROFILE_ID: VSLICKXNOPLUSNEWMSG=1 XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop BROWSER=chrome PATH=/home/username/.venv312/bin:/home/username/tools:/home/username/.local/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/snap/bin:/opt/slickedit-pro2023/bin:/opt/google/chrome DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus HOSTTYPE=x86_64 POWERLINE_COMMAND=oh-my-posh TOOLSDIR=/home/username/tools WT_PROFILE_ID={e5a83caa-4c73-52b3-ae6b-bc438d721ef9} PLANTUML_JAR_PATH=/usr/share/plantuml _=/home/linuxbrew/.linuxbrew/bin/oh-my-posh]
[TRACE] 09:38:35.402 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.402 shell.go:Pwd:428 → /home/username
[TRACE] 09:38:35.402 shell.go:Pwd() - 13µs
[TRACE] 09:38:35.402 shell.go:GOOS() - 100ns
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[DEBUG] 09:38:35.402 shell.go:User:563 → username
[TRACE] 09:38:35.402 shell.go:User() - 2.7µs
[DEBUG] 09:38:35.402 shell.go:Host:575 → COMPUTER
[TRACE] 09:38:35.402 shell.go:Host() - 4.1µs
[TRACE] 09:38:35.402 shell.go:GOOS() - 0s
[DEBUG] 09:38:35.402 shell_unix.go:Platform:87 → ubuntu
[DEBUG] 09:38:35.402 shell.go:Getenv:398 → 1
[TRACE] 09:38:35.402 shell.go:Getenv(SHLVL) - 2.4µs
[TRACE] 09:38:35.402 shell.go:TemplateCache() - 118.3µs
[DEBUG] 09:38:35.402 shell.go:Getenv:398 → 39
[TRACE] 09:38:35.402 shell.go:Getenv(POSH_CURSOR_LINE) - 2.9µs
[DEBUG] 09:38:35.402 shell.go:Getenv:398 → 1
[TRACE] 09:38:35.402 shell.go:Getenv(POSH_CURSOR_COLUMN) - 2.4µs
[TRACE] 09:38:35.402 shell.go:Flags() - 100ns
[TRACE] 09:38:35.402 shell.go:Flags() - 0s
[TRACE] 09:38:35.402 shell.go:Shell() - 0s
[TRACE] 09:38:35.402 shell.go:Shell() - 0s
[TRACE] 09:38:35.402 shell.go:Flags() - 100ns
[TRACE] 09:38:35.403 shell.go:Pwd() - 200ns
[TRACE] 09:38:35.403 shell.go:GOOS() - 0s
[DEBUG] 09:38:35.403 segment.go:SetEnabled:523 → Segment: Text
[TRACE] 09:38:35.403 shell.go:TemplateCache() - 100ns
[TRACE] 09:38:35.403 shell.go:Flags() - 100ns
[TRACE] 09:38:35.403 shell.go:Flags() - 200ns
[TRACE] 09:38:35.403 shell.go:Pwd() - 0s
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.403 segment.go:SetEnabled:523 → Segment: Path
[TRACE] 09:38:35.403 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.403 properties.go:GetString:28 → environment
[TRACE] 09:38:35.403 shell.go:Shell() - 100ns
[TRACE] 09:38:35.403 shell.go:Shell() - 100ns
[TRACE] 09:38:35.403 shell.go:Pwd() - 100ns
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[TRACE] 09:38:35.403 shell.go:Flags() - 200ns
[DEBUG] 09:38:35.403 segment.go:SetEnabled:523 → Segment: Python
[DEBUG] 09:38:35.403 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 09:38:35.403 properties.go:GetString:28 → {{ .Code }}
[DEBUG] 09:38:35.403 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 09:38:35.403 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv venv-win pyenv-win]
[DEBUG] 09:38:35.403 properties.go:GetString:28 → 
[DEBUG] 09:38:35.403 properties.go:GetBool:22 → home_enabled: false
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[TRACE] 09:38:35.403 shell.go:Pwd() - 200ns
[TRACE] 09:38:35.403 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.403 properties.go:GetString:28 → ~
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.403 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 09:38:35.403 shell.go:Flags() - 100ns
[TRACE] 09:38:35.403 shell.go:PathSeparator() - 100ns
[TRACE] 09:38:35.403 shell.go:Pwd() - 100ns
[TRACE] 09:38:35.403 shell.go:GOOS() - 0s
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.403 segment.go:SetEnabled:523 → Segment: Status
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.403 segment.go:SetEnabled:523 → Segment: Session
[DEBUG] 09:38:35.403 shell.go:Getenv:398 → NO DATA
[TRACE] 09:38:35.403 shell.go:StatusCodes() - 300ns
[TRACE] 09:38:35.403 shell.go:Getenv(SSH_CONNECTION) - 2.7µs
[DEBUG] 09:38:35.403 shell.go:Getenv:398 → NO DATA
[DEBUG] 09:38:35.403 text.go:Render:70 → Rendering template: {{ .Code }}
[TRACE] 09:38:35.403 shell.go:Getenv(SSH_CLIENT) - 2.1µs
[DEBUG] 09:38:35.403 shell_unix.go:Platform:87 → ubuntu
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[TRACE] 09:38:35.403 shell.go:GOOS() - 0s
[TRACE] 09:38:35.403 shell.go:PathSeparator() - 0s
[TRACE] 09:38:35.403 shell.go:PathSeparator() - 0s
[TRACE] 09:38:35.403 shell.go:Flags() - 200ns
[TRACE] 09:38:35.403 shell.go:GOOS() - 0s
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[TRACE] 09:38:35.403 shell.go:Pwd() - 200ns
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.403 segment.go:SetEnabled:523 → Segment: Shell
[DEBUG] 09:38:35.403 properties.go:GetKeyValueMap:46 → mapped_shell_names: map[bash:SH pwsh:PS universal:Unknown Shell]
[TRACE] 09:38:35.403 shell.go:Shell() - 100ns
[TRACE] 09:38:35.403 shell.go:Flags() - 100ns
[TRACE] 09:38:35.403 shell.go:TemplateCache() - 100ns
[TRACE] 09:38:35.403 shell.go:PathSeparator() - 0s
[TRACE] 09:38:35.403 shell.go:GOOS() - 200ns
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[TRACE] 09:38:35.403 shell.go:PathSeparator() - 0s
[TRACE] 09:38:35.403 shell.go:GOOS() - 0s
[TRACE] 09:38:35.403 shell.go:GOOS() - 100ns
[TRACE] 09:38:35.403 shell.go:PathSeparator() - 0s
[TRACE] 09:38:35.403 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.403 shell_unix.go:IsWsl:35 → true
[TRACE] 09:38:35.403 shell_unix.go:IsWsl() - 6.1µs
[TRACE] 09:38:35.405 shell.go:TemplateCache() - 1µs
[DEBUG] 09:38:35.405 properties.go:GetBool:22 → always_enabled: false
[DEBUG] 09:38:35.405 shell.go:RunCommand:593 → NO DATA
[TRACE] 09:38:35.405 shell.go:RunCommand(who am i) - 1.965602ms
[DEBUG] 09:38:35.405 shell.go:RunCommand:593 → //wsl.localhost/Ubuntu-22.04/home/username
[TRACE] 09:38:35.405 shell.go:RunCommand(wslpath -m /home/username) - 1.527701ms
[TRACE] 09:38:35.405 shell.go:StackCount() - 300ns
[TRACE] 09:38:35.405 shell.go:TemplateCache() - 300ns
[TRACE] 09:38:35.405 shell_unix.go:DirIsWritable(/home/username) - 7µs
[TRACE] 09:38:35.405 shell.go:TemplateCache() - 100ns
[DEBUG] 09:38:35.405 text.go:Render:70 → Rendering template: {{ if .Env.PROJECT_INFO }}{{ .Env.PROJECT_INFO }}{{end}}
[DEBUG] 09:38:35.405 text.go:Render:70 → Rendering template: {{ .Name }} 
[DEBUG] 09:38:35.405 text.go:Render:70 → Rendering template: {{ .Path }}
[DEBUG] 09:38:35.405 text.go:Render:70 → Rendering template: ┌ {{ if .SSHSession }}SSH {{ end }}{{ .UserName }}@{{ .HostName }} 
[TRACE] 09:38:35.405 shell.go:TemplateCache() - 300ns
[TRACE] 09:38:35.405 shell.go:Shell() - 100ns
[TRACE] 09:38:35.405 shell.go:TemplateCache() - 300ns
[TRACE] 09:38:35.405 shell.go:Shell() - 100ns
[TRACE] 09:38:35.405 shell.go:TemplateCache() - 300ns
[TRACE] 09:38:35.405 shell.go:TemplateCache() - 200ns
[TRACE] 09:38:35.405 shell.go:Shell() - 100ns
[TRACE] 09:38:35.406 shell.go:TemplateCache() - 300ns
[TRACE] 09:38:35.406 shell.go:Shell() - 100ns
[DEBUG] 09:38:35.406 shell.go:Getenv:398 → NO DATA
[TRACE] 09:38:35.406 shell.go:Getenv(TERM_PROGRAM) - 11µs
[DEBUG] 09:38:35.406 text.go:Render:70 → Rendering template: plain
[DEBUG] 09:38:35.406 text.go:Render:70 → Rendering template: plain
[DEBUG] 09:38:35.406 text.go:Render:70 → Rendering template: plain
[DEBUG] 09:38:35.406 text.go:Render:70 → Rendering template: diamond
[DEBUG] 09:38:35.406 text.go:Render:70 → Rendering template: diamond
[DEBUG] 09:38:35.406 text.go:Render:70 → Rendering template: plain
[TRACE] 09:38:35.406 shell.go:Shell() - 200ns
[TRACE] 09:38:35.406 shell.go:Shell() - 100ns
[TRACE] 09:38:35.406 shell.go:Shell() - 0s
[TRACE] 09:38:35.406 shell.go:Shell() - 0s
[TRACE] 09:38:35.406 shell.go:Flags() - 100ns
[TRACE] 09:38:35.406 shell.go:Flags() - 200ns
[TRACE] 09:38:35.406 shell.go:Pwd() - 200ns
[TRACE] 09:38:35.406 shell.go:GOOS() - 0s
[DEBUG] 09:38:35.406 segment.go:SetEnabled:523 → Segment: Gitversion
[TRACE] 09:38:35.406 shell.go:Flags() - 200ns
[TRACE] 09:38:35.406 shell.go:Pwd() - 100ns
[TRACE] 09:38:35.406 shell.go:GOOS() - 0s
[DEBUG] 09:38:35.406 segment.go:SetEnabled:523 → Segment: Git
[DEBUG] 09:38:35.406 shell_unix.go:IsWsl:35 → true
[TRACE] 09:38:35.406 shell.go:Flags() - 200ns
[TRACE] 09:38:35.406 shell_unix.go:IsWsl() - 3.8µs
[TRACE] 09:38:35.406 shell.go:Pwd() - 100ns
[TRACE] 09:38:35.406 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.406 segment.go:SetEnabled:523 → Segment: Project
[TRACE] 09:38:35.406 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.406 shell.go:FileContent:537 ↓
    5.15.133.1-microsoft-standard-WSL2
    
[TRACE] 09:38:35.406 shell.go:Pwd() - 300ns
[TRACE] 09:38:35.406 shell.go:GOOS() - 100ns
[TRACE] 09:38:35.406 shell.go:FileContent(/proc/sys/kernel/osrelease) - 71.3µs
[TRACE] 09:38:35.406 shell_unix.go:IsWsl2() - 85µs
[DEBUG] 09:38:35.406 segment.go:SetEnabled:523 → Segment: Root
[TRACE] 09:38:35.406 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.406 shell.go:CommandPath:615 → /home/linuxbrew/.linuxbrew/bin/gitversion
[TRACE] 09:38:35.406 shell.go:Flags() - 200ns
[TRACE] 09:38:35.406 shell.go:CommandPath(gitversion) - 466.6µs
[TRACE] 09:38:35.406 shell_unix.go:Root() - 1.1µs
[TRACE] 09:38:35.406 shell.go:HasCommand(gitversion) - 472.5µs
[TRACE] 09:38:35.406 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.406 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.406 shell.go:HasFiles(package.json) - 81.9µs
[TRACE] 09:38:35.406 shell.go:GOOS() - 100ns
[TRACE] 09:38:35.406 shell.go:Pwd() - 100ns
[TRACE] 09:38:35.406 shell.go:Pwd() - 0s
[DEBUG] 09:38:35.406 segment.go:SetEnabled:523 → Segment: Svn
[DEBUG] 09:38:35.406 properties.go:GetInt:40 → cache_timeout: 30
[DEBUG] 09:38:35.406 shell_unix.go:IsWsl:35 → true
[TRACE] 09:38:35.406 shell_unix.go:IsWsl() - 5.7µs
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(Cargo.toml) - 42.2µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 0s
[DEBUG] 09:38:35.407 shell.go:FileContent:537 ↓
    5.15.133.1-microsoft-standard-WSL2
    
[TRACE] 09:38:35.407 shell.go:FileContent(/proc/sys/kernel/osrelease) - 51.7µs
[TRACE] 09:38:35.407 shell_unix.go:IsWsl2() - 64.2µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(pyproject.toml) - 70.7µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(composer.json) - 49µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(*.nuspec) - 140.8µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(*.sln) - 108.801µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 0s
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(*.slnf) - 105.4µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 0s
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(*.vbproj) - 71.4µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(*.fsproj) - 73.7µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 200ns
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(*.csproj) - 101.5µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 200ns
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(JuliaProject.toml) - 61.7µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.407 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.407 shell.go:HasFiles(Project.toml) - 61.1µs
[TRACE] 09:38:35.407 shell.go:Pwd() - 100ns
[DEBUG] 09:38:35.408 shell.go:HasFiles:464 → false
[TRACE] 09:38:35.408 shell.go:HasFiles(*.psd1) - 77.4µs
[DEBUG] 09:38:35.408 properties.go:GetBool:22 → always_enabled: false
[DEBUG] 09:38:35.408 shell.go:RunCommand:593 → //wsl.localhost/Ubuntu-22.04/home/username
[TRACE] 09:38:35.408 shell.go:RunCommand(wslpath -m /home/username) - 1.224401ms
[TRACE] 09:38:35.408 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.408 shell.go:RunCommand:593 → //wsl.localhost/Ubuntu-22.04/home/username
[TRACE] 09:38:35.408 shell.go:RunCommand(wslpath -m /home/username) - 1.343001ms
[TRACE] 09:38:35.408 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.408 shell.go:CommandPath:615 → /usr/bin/git
[TRACE] 09:38:35.408 shell.go:CommandPath(git) - 30.9µs
[ERROR] 09:38:35.408 shell.go:CommandPath:619 → exec: "svn": executable file not found in $PATH
[TRACE] 09:38:35.408 shell.go:HasCommand(git) - 33.9µs
[TRACE] 09:38:35.408 shell.go:Pwd() - 100ns
[TRACE] 09:38:35.408 shell.go:CommandPath(svn) - 57.6µs
[TRACE] 09:38:35.408 shell.go:HasCommand(svn) - 63.3µs
[ERROR] 09:38:35.408 shell.go:HasParentFilePath:759 → stat .git: no such file or directory
[TRACE] 09:38:35.408 shell.go:HasParentFilePath(.git) - 18.8µs
[DEBUG] 09:38:35.408 properties.go:GetBool:22 → fetch_bare_info: false
[ERROR] 09:38:35.636 shell.go:RunCommand:591 → exit status 1
[DEBUG] 09:38:35.636 shell.go:RunCommand:593 ↓
    Cannot find the .git directory
    
[TRACE] 09:38:35.636 shell.go:RunCommand(gitversion -output json) - 229.408272ms
[TRACE] 09:38:35.636 shell.go:Shell() - 300ns
[TRACE] 09:38:35.636 shell.go:Shell() - 100ns
[TRACE] 09:38:35.636 shell.go:Shell() - 0s
[TRACE] 09:38:35.636 shell.go:Shell() - 0s
[TRACE] 09:38:35.636 shell.go:Flags() - 100ns
[TRACE] 09:38:35.636 shell.go:Pwd() - 200ns
[TRACE] 09:38:35.636 shell.go:GOOS() - 100ns
[DEBUG] 09:38:35.636 segment.go:SetEnabled:523 → Segment: Text
[TRACE] 09:38:35.636 shell.go:TemplateCache() - 400ns
[TRACE] 09:38:35.636 shell.go:Flags() - 400ns
[TRACE] 09:38:35.636 shell.go:Pwd() - 100ns
[TRACE] 09:38:35.636 shell.go:GOOS() - 0s
[DEBUG] 09:38:35.636 segment.go:SetEnabled:523 → Segment: Text
[TRACE] 09:38:35.636 shell.go:TemplateCache() - 200ns
[DEBUG] 09:38:35.636 text.go:Render:70 → Rendering template:  $
[TRACE] 09:38:35.636 shell.go:Shell() - 200ns
[DEBUG] 09:38:35.636 text.go:Render:70 → Rendering template: └
[TRACE] 09:38:35.636 shell.go:Shell() - 200ns
[DEBUG] 09:38:35.636 shell.go:Getenv:398 → NO DATA
[TRACE] 09:38:35.636 shell.go:Getenv(TERM_PROGRAM) - 6.7µs
[DEBUG] 09:38:35.636 text.go:Render:70 → Rendering template: plain
[DEBUG] 09:38:35.636 text.go:Render:70 → Rendering template: plain
[TRACE] 09:38:35.636 shell.go:Shell() - 100ns
[TRACE] 09:38:35.636 shell.go:Shell() - 0s
[TRACE] 09:38:35.636 shell.go:Flags() - 0s
[TRACE] 09:38:35.636 shell.go:Shell() - 100ns
[TRACE] 09:38:35.636 shell.go:Shell() - 100ns
[DEBUG] 09:38:35.636 shell.go:Getenv:398 → NO DATA
[TRACE] 09:38:35.636 shell.go:Getenv(OMP_CACHE_DIR) - 3.4µs
[DEBUG] 09:38:35.636 shell.go:Getenv:398 → NO DATA
[TRACE] 09:38:35.636 shell.go:Getenv(XDG_CACHE_HOME) - 2.1µs
[TRACE] 09:38:35.636 shell_unix.go:CachePath() - 21.9µs
[TRACE] 09:38:35.636 shell.go:Flags() - 100ns
```

--- a/src/engine/block.go
+++ b/src/engine/block.go
@@ -151,7 +151,7 @@ func (b *Block) renderActiveSegment() {
 		b.writer.Write(ansi.Background, ansi.Foreground, b.activeSegment.text)
 	case Diamond:
 		background := ansi.Transparent
-		if b.previousActiveSegment != nil && !b.previousActiveSegment.hasTrailingDiamond() {
+		if b.previousActiveSegment != nil && b.previousActiveSegment.hasEmptyDiamondAtEnd() {
 			background = b.previousActiveSegment.background()
 		}
 		b.writer.Write(background, ansi.Background, b.activeSegment.LeadingDiamond)

--- a/src/engine/segment.go
+++ b/src/engine/segment.go
@@ -376,9 +376,12 @@ func (segment *Segment) isPowerline() bool {
 	return style == Powerline || style == Accordion
 }
 
-func (segment *Segment) hasTrailingDiamond() bool {
-	style := segment.style()
-	return style == Diamond && len(segment.TrailingDiamond) > 0
+func (segment *Segment) hasEmptyDiamondAtEnd() bool {
+	if segment.style() != Diamond {
+		return false
+	}
+
+	return len(segment.TrailingDiamond) == 0
 }
 
 func (segment *Segment) cwdIncluded() bool {

